### PR TITLE
Update `README` and add `LICENSE`

### PR DESCRIPTION
**Pull request by ddoktorski**
- Added missing `LICENSE` file
- Moved lcov example to a separate file
- Updated `README`

### CODE REVIEW

File: lcov.md
Comment (ddoktorski) [inline lcov.md:1]: Not sure if this is the best location for this file, but I think it makes sense to leave it somewhere in the repository.

File: README.md [hunk 1]
```diff
@@ -14,149 +19,64 @@ If you want to install a specific version, run the following command with the re
...
-> FN:3,8,calculator
-> FNDA:2,calculator
-> ... other metrics ...
-> LH:10
-> end_of_record
-> ```
+### Coverage statistics
+
+Various tools exist that can produce coverage statistics based on an lcov file. One of the most well-known among these is [genhtml](https://github.com/linux-test-project/lcov/blob/master/bin/genhtml),
+a tool that generates an html summary report using the coverage data in an lcov file, `genhtml` is part of the [lcov package](https://github.com/linux-test-project/lcov/tree/master).
```
Comment (ddoktorski) [inline README.md:54]: Not sure what approach we prefer here, whether we should simply inform users that there exist some tools and they need to find and use them, or whether we should provide some concrete examples.

File: README.md [hunk 2]
```diff
@@ -14,149 +19,64 @@ If you want to install a specific version, run the following command with the re
...
+
+## Usage in GitHub actions
+
+A variety of GitHub actions are available for analyzing coverage data for continuous integration purposes, which can accept input in the form of an lcov file.
+Examples of such actions include [CodeCov](https://github.com/codecov/codecov-action) and [Coveralls](https://github.com/coverallsapp/github-action).
+
+The example workflow below illustrates how to use coverage report generated by `snforge` in conjunction with `CodeCov`:
+
+```yaml
+name: Example cairo coverage workflow
```
Comment (ddoktorski) [inline README.md:64]: Not tested yet. Also using `snforge` in this example might be a bit confusing. On the other hand I guess this represents the actual use case for most users.

**Review by ksew1 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -14,149 +19,64 @@ If you want to install a specific version, run the following command with the re
...
+For a practical example of how this format appears in a file, consider examining [this json file](./crates/cairo-coverage/tests/data/test_project/snfoundry_trace/tests::test_call::my_test.json).
 
-```rust
-pub enum Operation {
-    Add,
-    Multiply,
-}
+## Integrated tools
 
+[//]: # (TODO: Update the link below)
```
Comment (Arcticae) [inline README.md:31]: TODO

File: README.md [hunk 4]
```diff
@@ -14,149 +19,64 @@ If you want to install a specific version, run the following command with the re
...
 ```
 
-## Example
+## External tools integration
 
-Let's say you have a following program
+`cairo-coverage` is tool-agnostic which means that it accepts input from any tool. However, these tools need to generate trace data in a specific expected format -
+the same format which is accepted by the [cairo-profiler](https://github.com/software-mansion/cairo-profiler/tree/main).
+For the exact code implementation of this format, please refer to [this page](https://github.com/software-mansion/cairo-profiler/blob/main/crates/trace-data/src/lib.rs).
+For a practical example of how this format appears in a file, consider examining [this json file](./crates/cairo-coverage/tests/data/test_project/snfoundry_trace/tests::test_call::my_test.json).
```
Comment (Arcticae) [inline README.md:27]: This is not really practical info 🤔 I think perhaps we should leave it for the end?

Comment (Arcticae) [reply README.md:27]: Maybe mentioning that we in fact do plan to support `cairo-test`?

File: README.md [hunk 5]
```diff
@@ -14,149 +19,64 @@ If you want to install a specific version, run the following command with the re
...
-    }
-}
+The generated output file is in the `lcov` format. For your convenience, you can find an explanation along with a simple example of the `lcov` format [here](./lcov.md).
 
-```
+### Generate coverage report
 
-and you cover it with tests
+To generate the report, run `cairo-coverage` with one or more `<PATH_TO_TRACE_DATA>` arguments, which specify the paths to the json files containing the trace data to be used for generating the coverage report.
+Optionally, you can provide the path to the output file using `--output-path <OUTPUT_PATH>`. If not specified, the output file will default to being saved as `coverage.lcov`.
```
Comment (Arcticae) [inline README.md:43]: This should be before output format

File: README.md [hunk 6]
```diff
@@ -14,149 +19,64 @@ If you want to install a specific version, run the following command with the re
...
-> ... other metrics ...
-> LH:10
-> end_of_record
-> ```
+### Coverage statistics
+
+Various tools exist that can produce coverage statistics based on an lcov file. One of the most well-known among these is [genhtml](https://github.com/linux-test-project/lcov/blob/master/bin/genhtml),
+a tool that generates an html summary report using the coverage data in an lcov file, `genhtml` is part of the [lcov package](https://github.com/linux-test-project/lcov/tree/master).
+
+## Usage in GitHub actions
```
Comment (Arcticae) [inline README.md:56]: This should be much, much earlier in the readme
This is the way most people will use it

File: README.md [hunk 7]
```diff
@@ -14,149 +19,64 @@ If you want to install a specific version, run the following command with the re
 curl -L https://raw.githubusercontent.com/software-mansion/cairo-coverage/main/scripts/install.sh | sh -s -- v0.1.0
 ```
```
Comment (Arcticae) [inline README.md:20]: There should be usage here immediately

File: lcov.md
Comment (Arcticae) [reply lcov.md:1]: This should be linked somewhere if a user is meant to read it

**Review by Arcticae — APPROVED** [review]

**Pull request merged**